src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -18,124 +18,106 @@ public partial class DeflateStream : Stream
         private Inflater _inflater;
         private Deflater _deflater;
         private byte[] _buffer;
-
         private int _asyncOperations;
-
-        private IFileFormatWriter _formatWriter;
-        private bool _wroteHeader;
         private bool _wroteBytes;
 
-        public DeflateStream(Stream stream, CompressionMode mode)
-            : this(stream, mode, false)
+        #region Public Constructors
+
+        public DeflateStream(Stream stream, CompressionMode mode): this(stream, mode, false, ZLibNative.Deflate_DefaultWindowBits)
```

Comment:
I went back-and-forth on this a few times. On one hand I like minimizing indirection to make looking through the code simpler, but on the other hand it's a bit more verbose. I'm pretty much directly on the fence as to which is better, so I'll go with your preference.